OneCommunityGlobal/HighestGoodNetworkApp
# Vishala Add Materials Page

# Description
Add Materials Page
Or Implements # (WBS) 3.3.7 (Phase 2)

## Related PR:
This frontend PR is related to the [#667](https://github.com/OneCommunityGlobal/HGNRest/pull/667) backend PR.

## Main changes explained:
- Created Add Materials Component
- Display of all materials
- Two ways to add units : a. Choose measurement from the list ,  b. Create new measurement 
- When measurement is newly added , some similarity checks are done.

## How to test:
1. check into current branch
2. do  `npm run start:local` to run this PR locally
3. Go to http://localhost:3000/bmdashboard/materials/add
4. Test for a presentable UI 
5. Page displays the list of materials to give users an overview before inserting the new material. Test for the materials display.
6. Test for 2 types of measurements insertion. Either select from the list or insert a new measurement.
7. Inserting a new measurement displays the similar measurements that are already available in the db if the similarity percentage is greater than 50%
8. Inserting a new measurement inserts the unit in units collection and displays it in the list of measurement units on the Add Materials Page
9. Editing one of the field (between new/old measurement) clears the other one restricting the user to add the measurement in only one way
10. Test for client side validations on the fields

## Screenshots or videos of changes:
[Untitled_ Dec 22,2023 9_17 PM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/32354ae4-7e12-463b-9354-afc73b5a6fd8)
